libcudacxx/include/cuda/std/__cuda/semaphore.h

```diff
@@ -12,6 +12,7 @@
 #define _LIBCUDACXX___CUDA_SEMAPHORE_H
 
 #include <cuda/std/detail/__config>
+#include <cuda/atomic>
```

Comment:
That include would need to be after the system header include guard.

Also including all of `cuda/atomic` is a bit overkill. Please include only the subheader you need